### PR TITLE
Rename project to sonic-client

### LINKED ISSUE #13: Rename project
The current name of the project seems a bit long, it'd be good to rename it to something like `sonic-client`.

### PR DESCRIPTION
**Pull request by edgarlatorre**
## ✍️ Description

As ElixirSonicClient was a bit long I'm changing it to SonicClient. After this is merged we can rename the project on GitHub and update our remote repo locally.  And hopefully everything would be working 😄 

This PR was created from the branch #12 

## ✅ Fixes

- Closes #13

### CODE REVIEW

**Review by aadroher — COMMENTED** [review]
The diff in this PR seems a bit off, as it includes changes in `.github/workflows/ci.yml`. What about merging `main` into it and see if it cleans up?

File: test/sonic_client_test.exs [hunk 1]
```diff
@@ -16,27 +16,27 @@ defmodule ElixirSonicClientTest do
...
-    assert {:ok, "PONG"} == ElixirSonicClient.ping(conn)
+    assert {:ok, "PONG"} == SonicClient.ping(conn)
     TcpConnection.close(conn)
   end
 
   test "stop connection" do
     {:ok, conn} =
-      ElixirSonicClient.start(
+      SonicClient.start(
         Kernel.to_charlist("sonic"),
```
Comment (aadroher) [inline test/sonic_client_test.exs]: Did we forget to replace this with a call to `host()`, here?

Comment (edgarlatorre) [reply test/sonic_client_test.exs]: This was from your branch, and this branch wasn't up to date with main

Comment (aadroher) [reply test/sonic_client_test.exs]: ✌️

Comment (aadroher) [reply test/sonic_client_test.exs]: I suspected this, this is why I used the "we" word. 😬

**Review by aadroher — APPROVED** [review]
Looks good! 👍

**Pull request merged**